metadata-ingestion/src/datahub/ingestion/source/unity/source.py

```diff
@@ -156,7 +156,7 @@ def create(cls, config_dict, ctx):
         config = UnityCatalogSourceConfig.parse_obj(config_dict)
         return cls(ctx=ctx, config=config)
 
-    def get_platform_instance_id(self) -> str:
+    def get_platform_instance_id(self) -> Optional[str]:
```

Comment:
These changes aren't strictly necessary, since mypy allows an override method to narrow the return type